beamer/agent/middleware.py

```diff
@@ -282,3 +282,27 @@ def rate_limiter(
 ) -> Callable[[RPCEndpoint, Any], RPCResponse]:
     state = _RateLimiterState()
     return functools.partial(_rate_limiter, make_request=make_request, w3=w3, state=state)
+
+
+def base_fee_setter(
+    make_request: _MakeRequest, _w3: Web3
+) -> Callable[[RPCEndpoint, Any], RPCResponse]:
+
+    current_fees = make_request(RPCEndpoint("eth_feeHistory"), ["0x1", "latest", [10, 100]])[
```

Comment:
question: Wouldn't everything outside of the inner middleware function only be called once? 
If so we have to move the fee fetching part into the inner middleware function

Reply:
i mean when i print out the values max fee is definetely changing and priority fee is just constant in ganache.